components/sync/engine_impl/brave_model_type_worker.cc

```diff
@@ -0,0 +1,111 @@
+// Allow re-enable types not often than once in 30 minutes
+base::TimeDelta kMinimalTimeBetweenResetMarker =
+    base::TimeDelta::FromMinutes(30);
+}  // namespace
+
+BraveModelTypeWorker::BraveModelTypeWorker(
+    ModelType type,
+    const sync_pb::ModelTypeState& initial_state,
+    bool trigger_initial_sync,
+    std::unique_ptr<Cryptographer> cryptographer,
+    PassphraseType passphrase_type,
+    NudgeHandler* nudge_handler,
+    std::unique_ptr<ModelTypeProcessor> model_type_processor,
+    CancelationSignal* cancelation_signal)
+    : ModelTypeWorker(type,
+                      initial_state,
+                      trigger_initial_sync,
+                      std::move(cryptographer),
+                      passphrase_type,
+                      nudge_handler,
+                      std::move(model_type_processor),
+                      cancelation_signal) {}
+
+BraveModelTypeWorker::~BraveModelTypeWorker() {}
+
+void BraveModelTypeWorker::OnCommitResponse(
+    const CommitResponseDataList& committed_response_list,
+    const FailedCommitResponseDataList& error_response_list) {
+  ModelTypeWorker::OnCommitResponse(committed_response_list,
+                                    error_response_list);
+
+  if (IsResetProgressMarkerRequired(error_response_list)) {
+    ResetProgressMarker();
+  }
+}
+
+// static
+size_t BraveModelTypeWorker::GetFailuresToResetMarkerForTests() {
+  return kFailuresToResetMarker;
+}
+
+// static
+base::TimeDelta BraveModelTypeWorker::MinimalTimeBetweenResetForTests() {
+  return kMinimalTimeBetweenResetMarker;
+}
+
+bool BraveModelTypeWorker::IsResetProgressMarkerRequired(
+    const FailedCommitResponseDataList& error_response_list) {
+  if (!last_reset_marker_time_.is_null() &&
+      base::Time::Now() - last_reset_marker_time_ <
+          kMinimalTimeBetweenResetMarker) {
+    // Reset progress marker due to 7th failure happening twice in a row
+    // in less than 30mins
+    // P3A sample is 2
+    UMA_HISTOGRAM_EXACT_LINEAR("Brave.Sync.ProgressTokenEverReset", 2, 2);
+    return false;
+  }
+
+  bool found_conflict_or_transient = false;
+  for (const syncer::FailedCommitResponseData& failed_response_entry :
+       error_response_list) {
+    if (failed_response_entry.response_type ==
+            sync_pb::CommitResponse_ResponseType_CONFLICT ||
+        failed_response_entry.response_type ==
+            sync_pb::CommitResponse_ResponseType_TRANSIENT_ERROR) {
+      found_conflict_or_transient = true;
+      break;
+    }
+  }
+
+  if (found_conflict_or_transient) {
+    ++failed_commit_times_;
+  } else {
+    failed_commit_times_ = 0;
+  }
+
+  return failed_commit_times_ >= kFailuresToResetMarker;
+}
+
+void BraveModelTypeWorker::ResetProgressMarker() {
+  VLOG(1) << "Reset progress marker for type " << ModelTypeToString(type_);
+  // Normal reset of progress marker due to 7th failure
+  // P3A sample is 1
+  UMA_HISTOGRAM_EXACT_LINEAR("Brave.Sync.ProgressTokenEverReset", 1, 2);
```

Comment:
we typically start with 0, so this should be 
`UMA_HISTOGRAM_EXACT_LINEAR("Brave.Sync.ProgressTokenEverReset", 0, 1);`
and the other one
`UMA_HISTOGRAM_EXACT_LINEAR("Brave.Sync.ProgressTokenEverReset", 1, 1);`

also using the macro twice is discouraged in chromium, so it would be better to wrap it into a function or use `UmaHistogramExactLinear` from histogram_functions.h

Reply:
also it would be needed to update https://github.com/brave/brave-browser/wiki/P3A once merged

Reply:
thanks, fixed at https://github.com/brave/brave-core/pull/8201/commits/11fbac0c323baa1e9c9d0ffac8af49d1c29534a9